### DISCUSSION TITLE
btnMatrix, setting the state

### DISCUSSION THREAD

**Discussion by matsekberg** [HASwitchPlate/openHASP#703] [Q&A]
Is there a way to set the state of individual buttons in a button matrix using MQTT?

And how can i get the state of each button? I only get up and down events, nothing about state of a toggle button.

**Comment by smcgann99** [comment]
When you press a button in the matrix you see the index of the button pressed e.g. event: "up" val: 0

**Reply by matsekberg** [reply]
Yes but I mean the other way around. Since they can be declared as toggle buttons there may be a need to set their state at startup for example
